File: KotlinAndroidTestInJava/app/src/main/java/nl/ckyong/kotlinandroidtest/HelloNameActivity.java
package nl.ckyong.kotlinandroidtest;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

/**
 * Created by ckyoung on 13-Jul-17.
 */
public class HelloNameActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_helloname);
        String name = getIntent().getExtras().get("nameText").toString();
        TextView nameText = (TextView) findViewById(R.id.nameDisplay);
        nameText.setText("Hello " + name + "!");
    }
}

File: KotlinAndroidTest/app/src/main/java/nl/ckyong/kotlinandroidtest/HelloNameActivity.kt
package nl.ckyong.kotlinandroidtest

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.TextView

/**
 * Created by ckyoung on 11-Jul-17.
 */

class HelloNameActivity : AppCompatActivity(){
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_helloname)
        val name:String = intent.extras["nameText"].toString()
        val nameText = findViewById(R.id.nameDisplay) as TextView
        nameText.text = "Hello $name!"
    }
}
